src/phoenix/db/bulk_inserter.py

```diff
@@ -86,28 +91,60 @@ def last_updated_at(self, project_rowid: Optional[ProjectRowId] = None) -> Optio
 
     async def __aenter__(
         self,
-    ) -> Tuple[Callable[[Span, str], Awaitable[None]], Callable[[pb.Evaluation], Awaitable[None]]]:
+    ) -> Tuple[
+        Callable[[Span, str], Awaitable[None]],
+        Callable[[pb.Evaluation], Awaitable[None]],
+        Callable[[DataModification], Awaitable[None]],
+    ]:
         self._running = True
+        self._jobs = Queue()
         self._task = asyncio.create_task(self._bulk_insert())
-        return self._queue_span, self._queue_evaluation
+        return (
+            self._queue_span,
+            self._queue_evaluation,
+            self._enqueue_for_transaction,
+        )
 
     async def __aexit__(self, *args: Any) -> None:
+        self._jobs = None
         self._running = False
 
+    async def _enqueue_for_transaction(self, job: DataModification) -> None:
+        await cast("Queue[DataModification]", self._jobs).put(job)
+
     async def _queue_span(self, span: Span, project_name: str) -> None:
         self._spans.append((span, project_name))
 
     async def _queue_evaluation(self, evaluation: pb.Evaluation) -> None:
         self._evaluations.append(evaluation)
 
+    async def _process_events(self, events: Iterable[Optional[DataModificationEvent]]) -> None: ...
+
     async def _bulk_insert(self) -> None:
+        assert isinstance(self._jobs, Queue)
```

Comment:
Finding it slightly awkward to have the optional type on `self._jobs`. Can we just have it be a queue type that we reset on `__aexit__`?

Reply:
see my other comment